src/scripts/store.js

```diff
@@ -321,9 +72,43 @@ function convertToArray(obj) {
   });
 }
 
-export function getStoredList() {
-  const storedListLS = localStorage.getItem("destinations");
+//check/uncheck an item from a list
+export async function toggleButtonVisited(item) {
+  const visited = item.visited;
+  item.visited = !visited;
+
+  const query = new Parse.Query(Destinations);
+  query.equalTo("city", item.city);
+  const updateQuery = await query.find();
+
+  updateQuery[0]
+    .set("visited", item.visited)
+    .save()
+    .then(
+      (myObject) => {
+        console.log("The object was updated successfully.");
+      },
+      (error) => {
+        console.log(error);
+      }
+    );
+}
 
-  const storedList = storedListLS ? JSON.parse(storedListLS) : [];
-  return convertToArray(storedList);
+//delete an item
+export async function deleteItem(item) {
+  const query = new Parse.Query(Destinations);
+  query.equalTo("city", item.city);
+  const deleteQuery = await query.find();
+
+  if (confirm("Are you sure?")) {
+    deleteQuery[0].destroy().then(
+      (myObject) => {
+        console.log("The object was deleted successfully.");
+      },
+      (error) => {
+        console.log(error);
+      }
+    );
```

Comment:
Did you understood the diff between promise/then vs async/await? You can turn any async code into async/await. For example this would be:

```js
try {
  const object = await deleteQuery[0].destroy();
  console.log("The object was deleted successfully.", object);
} catch (e) {
  console.log("Delete failed!", error);
}
```

Simpler, right? You can do the same for other code blocks!

Reply:
Awesome, I will!